pkg/builder/local_build_executor_test.go

```diff
@@ -325,13 +329,14 @@ func TestLocalBuildExecutorOutputSymlinkReadingFailure(t *testing.T) {
 	filePool := mock.NewMockFilePool(ctrl)
 	buildDirectory.EXPECT().InstallHooks(filePool)
 	buildDirectory.EXPECT().Mkdir("root", os.FileMode(0777))
-	inputRootDirectory := mock.NewMockBuildDirectory(ctrl)
 	buildDirectory.EXPECT().EnterBuildDirectory("root").Return(inputRootDirectory, nil)
 	inputRootDirectory.EXPECT().MergeDirectoryContents(
 		ctx,
 		digest.MustNewDigest("nintendo64", "7777777777777777777777777777777777777777777777777777777777777777", 42),
 	).Return(nil)
 	inputRootDirectory.EXPECT().Mkdir("foo", os.FileMode(0777)).Return(nil)
+	inputRootDirectory.EXPECT().EnterDirectory("foo").Return(fooDirectory, nil)
```

Comment:
These calls don't seem necessary, right?

Reply:
It's the createOutputDirectories function, which uses getDirectory just to create parents and then closes the result.

The alternative is duplicating most of getDirectory, which seems worse.